apis/v1alpha1/generator.yaml

```diff
@@ -72,6 +74,47 @@ resources:
         type: map[string]*string
     tags:
       ignore: true
+  InstanceProfile:
+    renames:
+      operations:
+        CreateInstanceProfile:
+          input_fields:
+            InstanceProfileName: Name
+        GetInstanceProfile:
+          input_fields:
+            InstanceProfileName: Name
+        DeleteInstanceProfile:
+          input_fields:
+            InstanceProfileName: Name
+    exceptions:
+      terminal_codes:
+        - InvalidInput
+    fields:
+      Path:
+        late_initialize: {}
+      # In order to support attaching a role to the instance profile, custom code
+      # is used to attach/detach the role to the instance profile object based on
+      # the name of the role. Note that an instance profile can only contain a
+      # single IAM role at a given time.
+      Role:
+        type: "*string"
+        references:
+          resource: Role
+          path: Spec.Name
```

Comment:
Thanks for explaining this. I wonder why the IAM API semantics kinda gives the impression that multiple roles, can be added to an instance profile. Maybe it's something they will support in the future?

Reply:
Yes, it's a little strange here because the output for the attached role is technically in an array format:
- https://github.com/aws/aws-sdk-go/blob/main/service/iam/api.go#L27612
But in the following document mentions `An instance profile can contain only one IAM role`:
- https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html

Perhaps you're right, they might support multiple in the future.